### PR TITLE
rail: Update to latest spec version 28.0

### PR DESCRIPTION
**Pull request by mfleisz**

### CODE REVIEW

File: channels/rail/client/rail_orders.c [hunk 1]
```diff
@@ -261,14 +261,14 @@ static UINT rail_read_server_get_appid_resp_order(wStream* s,
 	if (!s || !getAppidResp)
 		return ERROR_INVALID_PARAMETER;
 
-	if (Stream_GetRemainingLength(s) < 516)
+	if (Stream_GetRemainingLength(s) < 524)
```
Comment (akallabeth) [inline channels/rail/client/rail_orders.c]: sure this will not break older servers?

File: winpr/include/winpr/stream.h [hunk 2]
```diff
@@ -180,6 +180,12 @@ static INLINE void Stream_Write_UINT8(wStream* _s, UINT8 _v)
 	*_s->pointer++ = (UINT8)(_v);
 }
 
+static INLINE void Stream_Write_INT16(wStream* _s, INT16 _v)
```
Comment (akallabeth) [inline winpr/include/winpr/stream.h]: Not necessary, the `Stream_Write_UINT16`   does the same ;)

File: channels/rail/client/rail_orders.c (hunk 1 above)
Comment (mfleisz) [reply channels/rail/client/rail_orders.c]: To my understanding this was a documentation error and the PDU actually contained 520 bytes of application id data from the beginning. Inf fact the current spec is still erroneous as it still shows the PDU with 512 bytes in the Protocol Examples chapter.

File: winpr/include/winpr/stream.h (hunk 2 above)
Comment (mfleisz) [reply winpr/include/winpr/stream.h]: Yes but that takes a UINT16 and I don't wanted to introduce new warnings/casts

File: channels/rail/client/rail_orders.c (hunk 1 above)
Comment (mfleisz) [reply channels/rail/client/rail_orders.c]: Just checked legacy rail to a Win7 machine and it definitely sends a 520 bytes application id.

File: winpr/include/winpr/stream.h (hunk 2 above)
Comment (akallabeth) [reply winpr/include/winpr/stream.h]: true.

File: channels/rail/client/rail_orders.c (hunk 1 above)
Comment (akallabeth) [reply channels/rail/client/rail_orders.c]: ah, thanks for the clarification.

**Pull request merged**